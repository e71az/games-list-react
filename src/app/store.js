import { configureStore } from '@reduxjs/toolkit';
import recipeReducer from '../reducers/recipeSlice';

const store = configureStore({
  reducer: {
    recipe: recipeReducer,
  },
});

export default store;
