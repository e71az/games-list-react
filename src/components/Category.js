/* eslint-disable react/forbid-prop-types */
import { useDispatch } from 'react-redux';
import { Link } from 'react-router-dom';
import { PropTypes } from 'prop-types';
import { mealFilter } from '../reducers/recipeSlice';

const Category = ({ category: { strCategory, strCategoryThumb } }) => {
  const dispatch = useDispatch();

  return (
    <Link to="/meals" onClick={() => dispatch(mealFilter(strCategory))}>
      <div
        data-testid="category"
        className="img-category-container"
        style={{
          backgroundImage: `url(${strCategoryThumb})`,
        }}
      >
        <h5 className="transparent-banner pl-3 py-2">{strCategory}</h5>
      </div>
    </Link>
  );
};

Category.propTypes = {
  category: PropTypes.object.isRequired,
};

export default Category;
