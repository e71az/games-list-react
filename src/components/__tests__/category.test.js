import { render, screen, cleanup } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../../app/store';
import Category from '../Category';

afterEach(() => {
  cleanup();
});

describe('Meal component tests', () => {
  const category = {
    idCategory: 1,
    strCategory: 'Seafood',
    strCategoryThumb:
      'https://media-cdn.tripadvisor.com/media/photo-s/18/3a/09/6c/bonefish-seafood-platter.jpg',
    strCategoryDescription: 'Seafood is delicious!',
  };

  // This render is for normal testing
  render(
    <Provider store={store}>
      <Router>
        <Category category={category} />
      </Router>
    </Provider>,
  );

  // This renderer is for snapshot testing
  const tree = renderer
    .create(
      <Provider store={store}>
        <Router>
          <Category category={category} />
        </Router>
      </Provider>,
    )
    .toJSON();

  const categoryElement = screen.getByTestId('category');

  test('Component should Render', () => {
    expect(categoryElement).toBeInTheDocument();
  });
  test('Component should contain Seafood title', () => {
    expect(categoryElement).toHaveTextContent('Seafood');
  });
  test('Matches snapshot', () => {
    expect(tree).toMatchSnapshot();
  });
});
