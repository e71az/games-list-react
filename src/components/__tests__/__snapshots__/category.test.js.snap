// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Meal component tests Matches snapshot 1`] = `
<a
  href="/meals"
  onClick={[Function]}
>
  <div
    className="img-category-container"
    data-testid="category"
    style={
      Object {
        "backgroundImage": "url(https://media-cdn.tripadvisor.com/media/photo-s/18/3a/09/6c/bonefish-seafood-platter.jpg)",
      }
    }
  >
    <h5
      className="transparent-banner pl-3 py-2"
    >
      Seafood
    </h5>
  </div>
</a>
`;
